bestofjs/bestofjs-webui
# Add time-period sort picker on Homepage onto Hot Projects list

Resolves #270 

## Goal
Provide a quick way to check the trends from the last 7 days, the last 30 days and the last 12 months, right from the home page.

Please see the commit messages to follow the changes.

### Things to consider
1. I noticed the selector doesn't only change the hot projects list, but it also affects the "Featured" projects list. This behavior also occurs on the older website version https://classic.bestofjs.org/, I thought this behavior is misleading the user UI-wise, because the selector is within the "Hot projects" list container.
2. In the issue description, the naming convention for the URL is e.g. `/this-week` that would show the "Hot projects this week". However, I opted to use "weekly" in the PR to follow the same naming convention used for the sorting of other projects lists (e.g. https://bestofjs.org/projects?sort=weekly)

## How to test
- Run `pnpm dev` OR open the preview link posted in Vercel's bot msg below
- Try any of the options on the Hot Projects selector on homepage
- Notice the change of the projects in the list and the URL - the query string should follow the option selected. FYI: the default option shows the daily so `/` and `/?sort=daily` are the same

## Screenshots

<img width="1309" alt="Screenshot 2024-03-19 at 13 01 52" src="https://github.com/bestofjs/bestofjs/assets/11666431/5486c0ac-7856-4210-8049-1cefcf7e26a8">

---

<img width="1322" alt="Screenshot 2024-03-19 at 13 02 12" src="https://github.com/bestofjs/bestofjs/assets/11666431/54e69119-c6e9-45a9-9c18-beee47e3b991">
